pygments/lexers/lisp.py

```diff
@@ -2692,3 +2692,148 @@ class FennelLexer(RegexLexer):
+        'os/exit', 'os/getenv', 'os/link', 'os/lstat', 'os/mkdir', 'os/mktime',
+        'os/perm-int', 'os/perm-string', 'os/readlink', 'os/realpath', 'os/rename',
+        'os/rm', 'os/rmdir', 'os/setenv', 'os/shell', 'os/sleep', 'os/stat',
+        'os/symlink', 'os/time', 'os/touch', 'os/umask', 'os/which', 'pairs', 'parse',
+        'parser/byte', 'parser/clone', 'parser/consume', 'parser/eof',
+        'parser/error', 'parser/flush', 'parser/has-more', 'parser/insert',
+        'parser/new', 'parser/produce', 'parser/state', 'parser/status',
+        'parser/where', 'partial', 'partition', 'peg/compile', 'peg/match', 'pos?',
+        'postwalk', 'pp', 'prewalk', 'prin', 'prinf', 'print', 'printf', 'product', 'prompt',
+        'propagate', 'protect', 'put', 'put-in', 'quit', 'range', 'reduce', 'reduce2',
+        'repeat', 'repl', 'require', 'resume', 'return', 'reverse', 'reversed', 'root-env',
+        'run-context', 'scan-number', 'seq', 'setdyn', 'shortfn', 'signal', 'slice',
+        'slurp', 'some', 'sort', 'sort-by', 'sorted', 'sorted-by', 'spit', 'stderr', 'stdin',
+        'stdout', 'string', 'string/ascii-lower', 'string/ascii-upper',
+        'string/bytes', 'string/check-set', 'string/find', 'string/find-all',
+        'string/format', 'string/from-bytes', 'string/has-prefix?',
+        'string/has-suffix?', 'string/join', 'string/repeat', 'string/replace',
+        'string/replace-all', 'string/reverse', 'string/slice', 'string/split',
+        'string/trim', 'string/triml', 'string/trimr', 'string?', 'struct', 'struct?',
+        'sum', 'symbol', 'symbol?', 'table', 'table/clone', 'table/getproto',
+        'table/new', 'table/rawget', 'table/setproto', 'table/to-struct', 'table?',
+        'take', 'take-until', 'take-while', 'tarray/buffer', 'tarray/copy-bytes',
+        'tarray/length', 'tarray/new', 'tarray/properties', 'tarray/slice',
+        'tarray/swap-bytes', 'thread/close', 'thread/current', 'thread/new',
+        'thread/receive', 'thread/send', 'trace', 'tracev', 'true?', 'truthy?', 'try',
+        'tuple', 'tuple/brackets', 'tuple/setmap', 'tuple/slice',
+        'tuple/sourcemap', 'tuple/type', 'tuple?', 'type', 'unless', 'unmarshal',
+        'untrace', 'update', 'update-in', 'use', 'values', 'var-', 'varfn', 'varglobal',
+        'walk', 'walk-ind', 'walk-dict', 'when', 'when-let', 'when-with', 'with',
+        'with-dyns', 'with-syms', 'with-vars', 'yield', 'zero?', 'zipcoll'
+    )
+
+    # Like Fennel's (no leading digits), and added ;,.\'
+    valid_name = r'[a-zA-Z_!$%&*+/:<=>?@^~|;,\'\.-][\w!$%&*+/:<=>?@^~|;,\'\.-]*'
+
+    tokens = {
+        'root': [
+            # comment
+            (r'#.*$', Comment.Single),
+
+            (r'\s+', Text),
+
+            # number
+            (r'-?\d+\.\d+', Number.Float),
+            (r'-?\d+', Number.Integer),
+            (r'0[xX][a-fA-F0-9]+', Number.Hex),
+
+            # double-quoted string
+            (r'@?"(\\\\|\\"|[^"])*"', String),
```

Comment:
This is going to result in catastrophic backtracking (similar for the case below.) Please do two things to make sure we won't suffer from this:

* Add a test case with a long string with backslashes in a string. That should take a looong time to execute, potentially timeout.
* Change it to `[^\\"]` to avoid the backtracking (same below), the test should pass just fine now.

Reply:
Hi Anteru. Sorry for the delay. Thank you for investigating this. Can you please tell me where to add this test case? I'm not familiar with adding tests to pygments, and also don't see a tests/test_lisp.py file.

Reply:
Just create a new file then :) The way to write a test is to use the `testcase` formatter (see also the docs), here's an example of the exact same bug just in the JsonLexer which you can use as a blueprint: https://github.com/gerner/pygments/commit/ec777784e381f29968705a7a3df9db83a2c7736f